conformance/utils/http/http.go

```diff
@@ -286,8 +290,27 @@ func CompareRequest(cReq *roundtripper.CapturedRequest, cRes *roundtripper.Captu
 		if expected.RedirectRequest == nil {
 			return nil
 		}
-		if expected.RedirectRequest.Hostname != cRes.RedirectRequest.Hostname {
-			return fmt.Errorf("expected redirected hostname to be %s, got %s", expected.RedirectRequest.Hostname, cRes.RedirectRequest.Hostname)
+		if expected.RedirectRequest.Host == "" {
+			expected.RedirectRequest.Host = cRes.RedirectRequest.Host
+		}
+
+		if expected.RedirectRequest.Port == "" {
+			expected.RedirectRequest.Port = "80"
+		}
+
+		if expected.RedirectRequest.Scheme == "" {
+			expected.RedirectRequest.Scheme = "http"
+		}
```

Comment:
what do you think about these lines?
I am not 100% sure it is best to take these assumptions

Reply:
The default for scheme and port made me pause slightly. Unless I'm misunderstanding things we're setting defaults here, conventionally you would (almost?) never redirect _to_ `http` so I'm just wondering if we should flip those :thinking:

Reply:
The thing is that I have the checks down there. For example the line 312 - I am checking for scheme. 
so when you would run for example a redirect test for Port when `expected.RedirectRequest.Scheme` is an empty string (because the implementations should not bother to set Scheme in their expected struct unless they want to check it) then we would fail the scheme. Hence I added it. 
My understanding is the by default it is http.

For port it is interesting, I will probably need to change it. Now I have tested Path redirect (changes are not in this PR) with envoyGateway other than istio because istio does not yet support it and I noticed that the response is an empty port, so "80" is not good.

If we decide to set the default to "80" we need to also fill `cRes.RedirectRequest.Port` to "80" but not sure thats desired.

Reply:
The other option I see is setting it to what we get from the expected field if it is an empty string but that also feels a bit iffy - something like this:
```go
if expected.RedirectRequest.Port == "" {
  expected.RedirectRequest.Port = cRes.RedirectRequest.Port
}
if expected.RedirectRequest.Scheme == "" {
  expected.RedirectRequest.Scheme = cRes.RedirectRequest.Scheme
}
```

I do have the same this for expected.RedirectRequest.Path which I think to add something similar:
```go
if expected.RedirectRequest.Path == "" {
  expected.RedirectRequest.Path = cRes.RedirectRequest.Path
}
```

Reply:
@youngnick and @robscott curious as to your thoughts here.

Reply:
My preference is to avoid defaults here, specify what we care about in the test, and only check for that. In this case that would mean lots of repetitive config, but I think it would more clearly communicate what we actually expect from an implementation.

Reply:
I have made some refactoring - PTAL